compiler/src/org.graalvm.compiler.hotspot.aarch64.test/src/org/graalvm/compiler/hotspot/aarch64/test/UseBarriersForVolatileTest.java

```diff
@@ -0,0 +1,296 @@
+        }
+    }
+
+    @Test
+    public void testVarHandlesInSubJVM() throws IOException, InterruptedException {
+        // the criterion for success is that all 4 tests finish ok
+        Probe successProbe = new Probe("OK (2 tests)", 1);
+        Probe testReadProbe = new Probe("Testing: testReadSnippet", 1);
+        Probe testWriteProbe = new Probe("Testing: testWriteSnippet", 1);
+        List<Probe> probes = Arrays.asList(testReadProbe, testWriteProbe, successProbe);
+        List<String> extraOpts = Arrays.asList("-XX:+UseBarriersForVolatile");
+        // run the tests belonging to inner class Internal in a subordinate test JVM
+        testHelper(probes, extraOpts, Internal.class.getName());
+    }
+
+    private static final boolean VERBOSE = Boolean.getBoolean(UseBarriersForVolatileTest.class.getSimpleName() + ".verbose");
+
+    // the name of the JUnit runner class we expect to find on the original command line
+    private static final String JUNIT_MAIN_CLASS_NAME = "com.oracle.mxtool.junit.MxJUnitWrapper";
+
+    // the property used to inhibit execution of desired tests in the main
+    // JVM and enable execution of desired tests in the subordinate JVM
+    private static final String ENABLE_SUBORDINATE_TESTS_PROPERTY = UseBarriersForVolatileTest.class.getName() + ".enable.subordinate.tests";
+
+    /**
+     * Gets the command line used to start the current tests, including all VM arguments, the test
+     * runner class and its option flags. This can be used to spawn an identical test run for the
+     * secondary test class.
+     */
+    public static List<String> getVMTestCommandLine() {
+        List<String> args = getProcessCommandLine();
+        if (args == null) {
+            throw new InternalError();
+        } else {
+            int index = findJUnitTestFilesIndex(args);
+            return args.subList(0, index);
+        }
+    }
+
+    private static int findJUnitTestFilesIndex(List<String> commandLine) {
+        int i = 1; // Skip the java executable
+        // inlcude all args up to and including the test runner class
+        while (i < commandLine.size()) {
+            String s = commandLine.get(i);
+            if (JUNIT_MAIN_CLASS_NAME.equals(s)) {
+                break;
+            } else if (hasArg(s)) {
+                i += 2;
+            } else {
+                i++;
+            }
+        }
+        if (i == commandLine.size()) {
+            throw new InternalError();
+        }
+        i++;
+        // skip all option args to the test runner class
+        while (i < commandLine.size()) {
+            String s = commandLine.get(i);
+            if (s.charAt(0) != '-') {
+                // this index identifies the first test class or an @file spec
+                return i;
+            } else if (hasArg(s)) {
+                i += 2;
+            } else {
+                i++;
+            }
+        }
+        throw new InternalError();
```

Comment:
`"Could not find index of test class or @file argument"`

Reply:
Also this one